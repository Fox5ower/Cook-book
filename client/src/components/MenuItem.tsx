import React, { Component } from 'react'
import SkeletonLoader from 'tiny-skeleton-loader-react'
import SkeletonStyles from '../services/static/preloader.style.json'
import DishActions from './DishActions'

interface MyProps {
  _id: string
  name: string
  category: string
  method: string
  description: string
  engreediants: Array<string>
  image: string
  toggleSlider: Function
  setIndex: Function
  counter: number
}

interface MyState {
  mounting: boolean
}

class MenuItem extends Component<MyProps, MyState> {
  constructor(props: MyProps) {
    super(props)

    this.state = {
      mounting: false,
    }
  }

  dynamicBackground = {
    backgroundImage: 'url(' + this.props.image + ')',
  }

  UNSAFE_componentWillMount() {
    this.setState({
      mounting: true,
    })
  }

  componentDidMount() {
    setTimeout(() => {
      this.setState({
        mounting: false,
      })
    })
  }

  render() {
    return (
      <div className="menu__item" style={this.dynamicBackground}>
        <div
          className="clicker"
          style={{ height: '100%', zIndex: 2, cursor: 'pointer' }}
          onClick={() => {
            this.props.toggleSlider()
            this.props.setIndex(this.props.counter)
          }}
        >
          <SkeletonLoader
            style={
              this.state.mounting
                ? SkeletonStyles
                : { display: 'none', transition: 'all .3s' }
            }
          />
          <div className="menu__item__inner-container">
            <div className="memu__item__head-container">
              <span className="menu__item__name">{this.props.name}</span>
              <span className="menu__item__counter">{this.props.counter}</span>
            </div>
            <br />
            <hr />
            <span className="menu__item__description">
              {this.props.description}
            </span>
          </div>
        </div>
        <DishActions name={this.props.name} counter={this.props.counter} />
      </div>
    )
  }
}

export default MenuItem
