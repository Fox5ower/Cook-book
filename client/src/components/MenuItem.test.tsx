import React from 'react'
import renderer from 'react-test-renderer'
import { configure, render, shallow } from 'enzyme'
import MenuItem from './MenuItem'
import Adapter from 'enzyme-adapter-react-16'
import { cleanup } from '@testing-library/react'

configure({ adapter: new Adapter() })

afterEach(cleanup)

const fakeDish = {
  _id: '1',
  name: 'Test Name',
  category: 'Test Category',
  method: 'Test Method',
  description: 'Test Description',
  engreediants: ['Test 1', 'Test 2', 'Test 3'],
  language: 'English',
  image: '/test-image',
}

const toggleSlider = (): void => {}

const setIndex = (): void => {}

it('Should renders without crashing', () => {
  render(
    <MenuItem
      _id={fakeDish._id}
      name={fakeDish.name}
      category={fakeDish.category}
      method={fakeDish.method}
      description={fakeDish.description}
      engreediants={fakeDish.engreediants}
      image={fakeDish.image}
      toggleSlider={toggleSlider}
      counter={1}
      setIndex={setIndex}
    />
  )
})

it('Should renders correctly with given props', () => {
  const menuItem = shallow(
    <MenuItem
      _id={fakeDish._id}
      name={fakeDish.name}
      category={fakeDish.category}
      method={fakeDish.method}
      description={fakeDish.description}
      engreediants={fakeDish.engreediants}
      image={fakeDish.image}
      toggleSlider={toggleSlider}
      counter={1}
      setIndex={setIndex}
    />
  )

  const expectedOutput = `<span class="menu__item__name">Test Name</span>`

  const realOutput = menuItem.find('.menu__item__name').html()
  expect(realOutput).toEqual(expectedOutput)
})

it('Should match the snapshot', () => {
  const tree = renderer
    .create(
      <MenuItem
        _id={fakeDish._id}
        name={fakeDish.name}
        category={fakeDish.category}
        method={fakeDish.method}
        description={fakeDish.description}
        engreediants={fakeDish.engreediants}
        image={fakeDish.image}
        toggleSlider={toggleSlider}
        counter={1}
        setIndex={setIndex}
      />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
