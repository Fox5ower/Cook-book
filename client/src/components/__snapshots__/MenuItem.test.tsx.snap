// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
<div
  className="menu__item"
  style={
    Object {
      "backgroundImage": "url(/test-image)",
    }
  }
>
  <div
    className="clicker"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "height": "100%",
        "zIndex": 2,
      }
    }
  >
    <div
      className="styles_skeleton__2Mj1U"
      style={
        Object {
          "background": "#E9ECEF",
          "borderRadius": "0",
          "display": "block",
          "height": "100%",
          "left": "0",
          "position": "absolute",
          "top": "0",
          "transition": "all .3s",
          "width": "100%",
          "zIndex": "9999",
        }
      }
    >
      â€Œ
    </div>
    <div
      className="menu__item__inner-container"
    >
      <div
        className="memu__item__head-container"
      >
        <span
          className="menu__item__name"
        >
          Test Name
        </span>
        <span
          className="menu__item__counter"
        >
          1
        </span>
      </div>
      <br />
      <hr />
      <span
        className="menu__item__description"
      >
        Test Description
      </span>
    </div>
  </div>
  <div
    className="dish-actions__container"
  >
    <div
      className="rating-area"
    >
      <input
        checked={false}
        id="star-1-5"
        onChange={[Function]}
        type="radio"
      />
      <label
        htmlFor="star-1-5"
        title="5"
      />
      <input
        checked={false}
        id="star-1-4"
        onChange={[Function]}
        type="radio"
      />
      <label
        htmlFor="star-1-4"
        title="4"
      />
      <input
        checked={false}
        id="star-1-3"
        onChange={[Function]}
        type="radio"
      />
      <label
        htmlFor="star-1-3"
        title="3"
      />
      <input
        checked={false}
        id="star-1-2"
        onChange={[Function]}
        type="radio"
      />
      <label
        htmlFor="star-1-2"
        title="2"
      />
      <input
        checked={false}
        id="star-1-1"
        onChange={[Function]}
        type="radio"
      />
      <label
        htmlFor="star-1-1"
        title="1"
      />
    </div>
    <svg
      className="favourite-btn"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
      />
    </svg>
  </div>
</div>
`;
